src/rt.erl

```diff
@@ -1187,12 +1186,9 @@ join_cluster(Nodes) ->
             %% ok do a staged join and then commit it, this eliminates the
             %% large amount of redundant handoff done in a sequential join
             [staged_join(Node, Node1) || Node <- OtherNodes],
-            plan_and_commit(Node1),
-            try_nodes_ready(Nodes, 3, 500)
+            ?assertEqual(ok, wait_until(fun() -> ok == plan_and_commit(Node1, Nodes) end))
     end,
 
-    ?assertEqual(ok, wait_until_nodes_ready(Nodes)),
```

Comment:
So `wait_until_nodes_ready` is very similar to `try_nodes_ready`. Since we're now guaranteed to call `try_nodes_ready` after every iteration of `plan_and_commit`, I believe the additional call to `wait_until_nodes_ready` is redundant.